pandas/core/groupby/ops.py

```diff
@@ -608,9 +618,52 @@ def _ea_wrap_cython_operation(
             f"function is not implemented for this dtype: {values.dtype}"
         )
 
+    @final
+    def _masked_ea_wrap_cython_operation(
+        self,
+        kind: str,
+        values: BaseMaskedArray,
+        how: str,
+        axis: int,
+        min_count: int = -1,
+        **kwargs,
+    ) -> BaseMaskedArray:
+        """
+        Equivalent of `_ea_wrap_cython_operation`, but optimized for masked EA's
+        and cython algorithms which accept a mask.
+        """
+        orig_values = values
+
+        # isna just directly returns self._mask, so copy here to prevent
+        # modifying the original
+        mask = isna(values).copy()
+        arr = values._data
+
+        if is_integer_dtype(values.dtype) or is_bool_dtype(values.dtype):
```

Comment:
```suggestion
        if is_integer_dtype(arr.dtype) or is_bool_dtype(arr.dtype):
```

This is equivalent, but those functions are a bit faster for numpy dtypes (and in the end it's also `arr` that we ensure to have a float dtype).

Reply:
Thanks, that makes sense